nessie/jobs/generate_canvas_caliper_analytics.py

```diff
@@ -49,21 +73,36 @@ def run(self):
             app.logger.error('Caliper explode schema and table creation failed.')
             return False
 
-        app.logger.info('Verify if data exists in caliper explode tables')
-        redshift_response = redshift.fetch(f'select count(*) from {redshift_schema_lrs_external}.{canvas_caliper_explode_table}')
+        # Sanity-check event times from the latest Caliper batch against previously transformed event times.
+        def datetime_from_query(query):
```

Comment:
Why define this function inside of `run` instead of at the class level?

Reply:
It's similar to defining a proc or lambda in Ruby, except that Python functions-within-functions let you give it a more descriptive name. The convention with these nested functions seems to be a signal that you only expect it to be useful for the code that immediately follows, and not to be more generally applicable.

Reply:
Ah, makes sense. Thanks for the explain.